image-builder/image_builder.sh

```diff
@@ -57,12 +59,30 @@ EOT
 exit "${error}"
 }
 
-while getopts "ho:s:" opt
+# Maximum allowed size in MB for root disk
+MAX_IMG_SIZE_MB=2048
+
+FS_TYPE=${FS_TYPE:-"ext4"}
+
+# Maximum no of attempts to create a root disk before giving up
+MAX_ATTEMPTS=5
+
+ATTEMPT_NUM=0
+
+while getopts "ho:s:f:" opt
 do
 	case "$opt" in
 		h)	usage ;;
 		o)	IMAGE="${OPTARG}" ;;
-		s)	IMG_SIZE="${OPTARG}" ;;
+		s)	IMG_SIZE=${OPTARG}
+                        if [ ${IMG_SIZE} -lt 0 ]; then
+                                die "Image size has to be greater than 0 MB."
```

Comment:
I think we need some further checks here as if I specify a range of values to `-s` I see:

- `-s -1`
  ```
  ERROR: Image size has to be greater than 0 MB.
  ```

- `-s 0`
  ```
  Error: The device /image/kata-containers.img is so small that it cannot possibly store a file system or partition table.  Perhaps you selected the wrong device?
  ```

- `-s 1`
  ```
  Error: Can't have the end before the start! (start sector=1953 length=-1857)
  ```

- `-s 2`
  ```
  Warning: The resulting partition is not properly aligned for best performance: 1953s % 2048s != 0s                                                                                            
  /dev/loop3p1: Not enough space to build proposed filesystem while setting up superblock                                                                                                       
  ```

- `-s 3`
  ```
  INFO: Image created. Size: 78MB.
  ```

Reply:
@jodh-intel Thanks, that a good point. But I think we should track that work under a different PR `for enhancing the error handling for manual image size`. This PR is mainly focusing on introducing automatic size selection for images. 

If you are OK with it, I can create a new issue and work on that PR. What do you say?

Reply:
Hi @harche - wfm. @jcvenegas - are you happy with that?

Reply:
ping @jcvenegas.

Reply:
@jodh-intel  @harche no problem to continue in a follow up PR. Lets do it.